x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -119,6 +135,37 @@ function getSpanItem(span: Span): IWaterfallItemSpan {
   };
 }
 
+function getErrorItem(error: APMError): IWaterfallItemError {
+  return {
+    id: error.error.id,
+    parentId: error.parent?.id,
+    serviceName: error.service.name,
+    message: error.error.log?.message || error.error.exception?.[0]?.message,
```

Comment:
It looks like we are adding more and more to the top-level `waterfallItem`. Maybe we should strip it down to just the things that are absolutely required for displaying it on the timeline:

```ts
interface IWaterfallItemBase<T> {
  doc: T;

  /**
   * Duration in us
   */
  duration: number;

  /**
   * start timestamp in us
   */
  timestamp: number;

  /**
   * offset from first item in us
   */
  offset: number;

  /**
   * skew from timestamp in us
   */
  skew: number;
}

export type IWaterfallItem = IWaterfallItemBase<Transaction> | IWaterfallItemBase<Span>;
```

I don't know if this is a good approach. My intention is to make the interface as small as possible.

Reply:
@sqren is it fine if I do this change in another PR? then I can remove the `childIds`, `getTransactionById` and other properties which are not necessary.

Reply:
Sure